File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int maxi=0,mini=prices[0];
        for(int i=1;i<n;i++){
            if(prices[i]>mini){
                maxi=Math.max(maxi, prices[i]-mini);
            }
            else{
                mini=prices[i];
            }
        }
        return maxi;
    }
}
/**

**/

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini = prices[0];
        int maxi = INT_MIN;
        int ans=0;
        for(int i=1;i<prices.size();i++){
            if(prices[i]<prices[i-1]){
                maxi=prices[i];
            }
            maxi=max(maxi,prices[i]);
            mini=min(mini,prices[i]);
            ans= max(ans, maxi-mini);
        }
        return ans;
    }
};